node/src/command.rs

```diff
@@ -249,24 +247,18 @@ pub fn run() -> Result<()> {
             Ok(())
         }
         None => {
-            let runner = cli.create_runner(&*cli.run)?;
+            let runner = cli.create_runner(&cli.run.normalize())?;
 
             runner.run_node_until_exit(|config| {
                 async move {
-                    // TODO
-                    let key = sp_core::Pair::generate().0;
-
-                    let extension = chain_spec::Extensions::try_get(&*config.chain_spec);
-                    let relay_chain_id = extension.map(|e| e.relay_chain.clone());
-                    let para_id = extension.map(|e| e.para_id);
+                    let para_id = chain_spec::Extensions::try_get(&*config.chain_spec).map(|e| e.para_id);
 
                     let polkadot_cli = RelayChainCli::new(
-                        config.base_path.as_ref().map(|x| x.path().join("polkadot")),
-                        relay_chain_id,
+                        &config,
                         [RelayChainCli::executable_name().to_string()].iter().chain(cli.relaychain_args.iter()),
                     );
 
-                    let id = ParaId::from(cli.run.parachain_id.or(para_id).unwrap_or(200));
+                    let id = ParaId::from(cli.run.parachain_id.or(para_id).unwrap_or(2000));
```

Comment:
do we need to specifically request this Parachain ID?

Reply:
well this PR is for our parachain that connects to Kusama, so i'd imagine the default parachain id that it'd fallback to in the cli would be our DataHighway parachain id.  so far it displays the live Kusama parachain id's that are being used here https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkusama.api.onfinality.io%2Fpublic-ws#/parachains, and it looks like 2000 has already been taken by Karura, so i think we should use `expect("Missing ParaId")`. i don't know why they switched to a fallback to the Karura parachain id in what's supposed to be a "template".

Reply:
it appears that para id 2000 may be Rococo-specific, according to https://substrate.dev/cumulus-workshop/#/en/3-parachains/2-register, please double check in Parachain Technical room on Matrix if it applies to Kusama too

Reply:
please find out how to reserve a parachain id as specified here https://substrate.dev/cumulus-workshop/#/en/3-parachains/2-register?id=registration-transaction. it has a link to "paraID being reserved" but it returns 404, so i'd suggest asking in Parachain Technical room on Matrix how to do it and then create a PR to fix the broken link in their docs

Reply:
@ltfschoen Got some responses from community https://matrix.to/#/!WuksvCDImqYSxvNmua:matrix.parity.io/$14W9j8ksOWo3FekiHjEAu6yZoPzd1jSeZVC24ByapeI?via=matrix.parity.io&via=matrix.org&via=web3.foundation

https://matrix.to/#/!WuksvCDImqYSxvNmua:matrix.parity.io/$BCdTh0mP07KNasF-Njm6PSsnKPK1jBrbt9PwHqAEjWg?via=matrix.parity.io&via=matrix.org&via=web3.foundation

We can take any > 1999 & < max (if no one has already taken it).

Here is parachain guide https://substrate.dev/substrate-how-to-guides/docs/parachains/overview/

Here is guide to reserve paraId https://substrate.dev/substrate-how-to-guides/docs/parachains/c-registration/register-paraid